langcodes/__init__.py

```diff
@@ -703,25 +705,39 @@ def distance(self, supported: 'Language') -> int:
             desired_triple = ('und', 'Zzzz', 'ZZ')
         else:
             desired_complete = self.prefer_macrolanguage().maximize()
-            desired_triple = (
-                desired_complete.language,
-                desired_complete.script,
-                desired_complete.territory,
-            )
-
+            if ignore_script:
+                desired_triple = (
+                    desired_complete.language,
+                    None,
+                    desired_complete.territory,
+                )
+            else:
+                desired_triple = (
+                    desired_complete.language,
+                    desired_complete.script,
+                    desired_complete.territory,
+                )
```

Comment:
I'd prefer the following:

```suggestion
              desired_triple = (
                  desired_complete.language,
                  None if ignore_script else desired_complete.script,
                  desired_complete.territory,
              )
```